artifact/metadata.go

```diff
@@ -295,9 +295,48 @@ func (a *ArtifactDepends) UnmarshalJSON(b []byte) error {
 	return nil
 }
 
+type SoftwareVersion struct {
+	Name  string `json:"name"`
+	Value string `json:"value"`
+}
+
 type ArtifactProvides struct {
-	ArtifactName  string `json:"artifact_name"`
-	ArtifactGroup string `json:"artifact_group,omitempty"`
+	ArtifactName    string           `json:"artifact_name"`
+	ArtifactGroup   string           `json:"artifact_group,omitempty"`
+	SoftwareVersion *SoftwareVersion `json:"version,omitempty"`
```

Comment:
I realize that is was not specified clearly enough in the task: We have two different `artifact_provides`, one in the global header, and one which is per payload. It is the payload header we want to change here, because it is a freeform structure, unlike this one. It's not super obvious in the documentation, but you can look at the difference in [the global one](https://github.com/mendersoftware/mender-artifact/blob/master/Documentation/artifact-format-v3.md#artifact_provides) and [the payload one](https://github.com/mendersoftware/mender-artifact/blob/master/Documentation/artifact-format-v3.md#artifact_provides-1), which is inside `type-info` per payload.

Sorry, I missed this when you asked earlier in the chat. This will probably substantially change your PR, so I'm going to stop here for now.

Reply:
I reworked my PR to write the software version attribute in the type-info.